pos_accounting/src/storage/in_memory.rs

```diff
@@ -59,6 +59,69 @@ impl InMemoryPoSAccounting {
             delegation_data,
         }
     }
+
+    #[cfg(test)]
+    pub(crate) fn all_pool_data(&self) -> &BTreeMap<PoolId, PoolData> {
+        &self.pool_data
+    }
+
+    #[cfg(test)]
+    pub(crate) fn all_delegation_data(&self) -> &BTreeMap<DelegationId, DelegationData> {
+        &self.delegation_data
+    }
+
+    #[cfg(test)]
+    pub(crate) fn all_delegation_balances(&self) -> &BTreeMap<DelegationId, Amount> {
+        &self.delegation_balances
+    }
+
+    #[cfg(test)]
+    pub(crate) fn check_consistancy(&self) {
+        // pool_balance and pool_data must contain the same keys
+        assert_eq!(self.pool_balances.len(), self.pool_data.len());
+        let pool_balance_and_data_consistant =
+            self.pool_balances.keys().all(|key| self.pool_data.contains_key(key));
+        assert!(pool_balance_and_data_consistant);
+
+        // delegation_balance and delegation_data must contain the same keys
+        //
+        // Note: delegation_balances and delegation_data can have different length
+        // because zero balances are removed.
```

Comment:
Initially, we decided not to store zero balances in deltas and storage because it seemed redundant. But maybe we should reconsider this for consistency reasons as it would be easier to check the state. Thoughts?

Reply:
What's wrong with "if it's not availabe, it's zero"?

Reply:
It's impossible to know if it's "not available because it's zero" or "not available because of some error"

Reply:
The problem there is that we'll have these zeroes forever. Not a good deal.

What you're talking about is not an error, but an inconsistency. 

So we have to store tons of useless data because we can't do proper arithmetic. I think we can do proper arithmetic checks to ensure that non-existent is zero.